### PR TITLE
Fix 406 - Inject PACH_COMMIT_ID into job

### PR DESCRIPTION
**Pull request by sjezewski**
Simple implementation. Added it to the k8s manifest on job creation, and added a test.

### CODE REVIEW

**Comment by sjezewski** [comment]
Fixes #406

File: src/server/pps/server/api_server.go [hunk 1]
```diff
@@ -972,6 +972,12 @@ func job(jobInfo *persist.JobInfo) *extensions.Job {
 								Privileged: &trueVal, // god is this dumb
 							},
 							ImagePullPolicy: "IfNotPresent",
+							Env: []api.EnvVar{
+								api.EnvVar{
+									Name:  "PACH_COMMIT_ID",
+									Value: jobInfo.OutputCommit.ID,
+								},
+							},
```
Comment (jdoliner) [inline src/server/pps/server/api_server.go]: I figure we might as well include the input commits as well while we're at it. Maybe as `PACH_OUT_COMMIT_ID` or something of that ilk.

**Comment by derekchiang** [comment]
Reminder that we probably want some documentation stating the fact that the commit ID is available through an env variable.

**Comment by sjezewski** [comment]
Good call @derekchiang  ... although I'm not sure exactly where to put this documentation. It feels like this might be a good instance of something to go on a dev docs site.

Otherwise ... maybe go api docs ... on the createjob / createpipeline functions?

File: src/server/pfs/server/server_test.go [hunk 2]
```diff
@@ -81,6 +81,16 @@ func TestInvalidRepo(t *testing.T) {
...
+
+	require.NoError(t, client.CreateRepo("lenny"))
+	require.NoError(t, client.CreateRepo("lenny123"))
+	require.NoError(t, client.CreateRepo("lenny_123"))
+
+	require.YesError(t, client.CreateRepo("lenny-123"))
+	require.YesError(t, client.CreateRepo("lenny.123"))
+	require.YesError(t, client.CreateRepo("lenny:"))
+	require.YesError(t, client.CreateRepo("lenny,"))
+	require.YesError(t, client.CreateRepo("lenny#"))
```
Comment (jdoliner) [inline src/server/pfs/server/server_test.go]: Assuming this is supposed to be a reference to our canine friend it's actually leni... Needless to say this is not a real defect in this code review.

**Comment by jdoliner** [comment]
LGTM

**Pull request merged**